File: oop-2s/3lab/3lab.cpp
#include <iostream>
#include <cmath>

class Vector {
private:
    double x, y;

public:
    Vector() : x(0), y(0) {}
    Vector(double x, double y) : x(x), y(y) {}
    Vector(const Vector& other) : x(other.x), y(other.y) {}

    double length() const {
        return std::sqrt(x * x + y * y);
    }

    double getX() const { return x; }
    double getY() const { return y; }

    Vector operator*(double scalar) const {
        return Vector(x * scalar, y * scalar);
    }

    Vector operator-(const Vector& other) const {
        return Vector(x - other.x, y - other.y);
    }

    void print() const {
        std::cout << "Vector(" << x << ", " << y << ") Length: " << length() << std::endl;
    }
};

int main() {
    Vector V1, V2(3, 4), V3 = V2;
    V3 = V3 * 2;
    V1 = V3 - V2;

    std::cout << "V1: "; V1.print();
    std::cout << "V2: "; V2.print();
    std::cout << "V3: "; V3.print();

    return 0;
}

File: oop-2s/3lab/3lab.cs
using System;

class Vector {
    private double x, y;

    public Vector() {
        x = 0; y = 0;
    }

    public Vector(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public Vector(Vector other) {
        x = other.x;
        y = other.y;
    }

    public double Length() {
        return Math.Sqrt(x * x + y * y);
    }

    public double GetX() { return x; }
    public double GetY() { return y; }

    public static Vector operator *(Vector v, double scalar) {
        return new Vector(v.x * scalar, v.y * scalar);
    }

    public static Vector operator -(Vector v1, Vector v2) {
        return new Vector(v1.x - v2.x, v1.y - v2.y);
    }

    public void Print() {
        Console.WriteLine($"Vector({x}, {y}) Length: {Length()}");
    }
}

class Program {
    static void Main() {
        Vector V1 = new Vector();
        Vector V2 = new Vector(3, 4);
        Vector V3 = new Vector(V2);

        V3 = V3 * 2;
        V1 = V3 - V2;

        Console.Write("V1: "); V1.Print();
        Console.Write("V2: "); V2.Print();
        Console.Write("V3: "); V3.Print();
    }
}
